spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/SqlIdentifier.java

```diff
@@ -90,7 +94,7 @@ public String toString() {
 	 * @see IdentifierProcessing#NONE
 	 */
 	default String getReference() {
```

Comment:
Please give this a proper JavaDoc comment.

After talking to Mark I now again understand what the `getReference` vs `toSql` is about.

`toSql` is to be used when rendering an identifier in SQL, as in `select id from someTable` 

`getReference` is used when accessing a column in a `ResultSet` here we must not apply any quoting.

This should be reflected in the JavaDoc comments of the two methods.